Assignment 2.Rmd

```diff
@@ -85,15 +89,49 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.widt
 
 ## Part II
 
-In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+#In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+```{r}
+(a1<- count(D1, stid))
+(a2<- count(D1, tid))
+names(a2) <- c("stid","n")
+a3 <- rbind(a1,a2)
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender,vertex.size = a3$n+15)
+```
 
 
 
 ## Part III
 
-Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+#Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+#Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+library(readr)
+DC1 <- read_csv("hudk4050-classes.csv")
+library(dplyr)
+#install.packages("tidyr")
+library(tidyr)
+library(igraph)
+DC1
+DC2 <- unite(DC1, "First Name", "Last Name", col = name, sep = " ")
+DC3 <- DC2 %>% gather(classnum, classcode, `Class 1`, `Class 2`, `Class 3`, `Class 4`, `Class 5`, `Class 6`) %>% select(-c(classnum))
+DC3$classcode = gsub(" ", "", DC3$classcode)
+DC3 <- DC3 %>% filter(classcode != "HUDK4050") %>% filter(name != "ZIMO CHEN")
+DC3$Count = 1
+DC4 <- DC3 %>% spread(classcode,Count)
+row.names(DC4) = DC4$name
+DC4$name <- NULL
+DC4 = ifelse(is.na(DC4), 0, 1)
+DC5 = as.matrix(DC4)
+DC6 = DC5 %*% t(DC5)
+diag(DC6) <- NA
+g3 = graph.adjacency(DC6, mode = "undirected")
+plot(g3, layout=layout.fruchterman.reingold)
+
+which.max(degree(g3))
+which.max(betweenness(g3))
```

Comment:
Try to interpret the centrality measures within the context of this dataset. In other words, what does it mean to be the most central person in this network?